packages/jest-circus/src/run.js

```diff
@@ -160,7 +160,12 @@ const _callCircusTest = (
 
   return callAsyncCircusFn(test.fn, testContext, {isHook: false, timeout})
     .then(() => dispatch({name: 'test_fn_success', test}))
-    .catch(error => dispatch({error, name: 'test_fn_failure', test}));
+    .catch(message => {
+      if (/^(Error: )?Exceeded timeout/.test(message)) {
```

Comment:
Not crazy about string matching here, any alternatives?

Reply:
We might consider adding a property to the error we reject with, which can be set to `timeout`, `assertion` etc? Kind of like `code` for node core errors. Not sure

Reply:
seems fixable :D